### PR TITLE
[FEATURE] Compatibility changes for TYPO3 10.4

### PR DESCRIPTION
**Pull request by felixsemmler**

### CODE REVIEW

File: Classes/System/RestApi/RestApiClient.php [hunk 1]
```diff
@@ -151,8 +152,8 @@ protected function createRequestException(RestException $e, $requestMethod, $req
      */
     protected function getRestlerBuilder()
     {
-        $objectManager = GeneralUtility::makeInstance('TYPO3\\CMS\\Extbase\\Object\\ObjectManager');
-        return $objectManager->get('Aoe\\Restler\\System\\Restler\\Builder');
+        $objectManager = GeneralUtility::makeInstance(ObjectManager::class);
```
Comment (tomasnorre) [inline Classes/System/RestApi/RestApiClient.php:155]: ObjectManager->get is deprecated in TYPO3 10, and will be removed in TYPO3 11. 
So please use the GeneralUtility::makeInstance() directly for the RestlerBuild::class

File: Classes/System/RestApi/RestApiRequest.php [hunk 2]
```diff
@@ -151,8 +152,8 @@ public function executeRestApiRequest($requestMethod, $requestUri, $getData = nu
      */
     protected function getRestApiJsonFormat()
     {
-        $objectManager = GeneralUtility::makeInstance('TYPO3\\CMS\\Extbase\\Object\\ObjectManager');
-        return $objectManager->get('Aoe\\Restler\\System\\RestApi\\RestApiJsonFormat');
+        $objectManager = GeneralUtility::makeInstance(ObjectManager::class);
+        return $objectManager->get(RestApiJsonFormat::class);
```
Comment (tomasnorre) [inline Classes/System/RestApi/RestApiRequest.php:156]: See comment above regarding ObjectManager and GeneralUtility::makeInstance

File: Classes/Configuration/ExtensionConfiguration.php [hunk 3]
```diff
@@ -45,7 +45,7 @@ class ExtensionConfiguration implements SingletonInterface
     public function __construct()
     {
         if (class_exists('\TYPO3\CMS\Core\Configuration\ExtensionConfiguration')) {
-            $this->configuration = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Configuration\ExtensionConfiguration::class)->get('restler');
+            $this->configuration = GeneralUtility::makeInstance('TYPO3\CMS\Core\Configuration\ExtensionConfiguration')->get('restler');
```
Comment (tomasnorre) [inline Classes/Configuration/ExtensionConfiguration.php:48]: Why did you change that?

Comment (felixsemmler) [reply Classes/Configuration/ExtensionConfiguration.php]: Sorry, I thought using the class constant for an non existing class leads to an error in PHP. I reverted my changes.

**Review by TomSchenk — APPROVED** [review]

**Review by aoekrz — APPROVED** [review]
